File: app/src/test/java/activity/amigosecreto/db/GrupoModelTest.java
package activity.amigosecreto.db;

import org.junit.Test;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

import static org.junit.Assert.*;

public class GrupoModelTest {

    @Test
    public void construtor_padrao_campos_sao_nulos_e_zero() {
        Grupo g = new Grupo();
        assertEquals(0, g.getId());
        assertNull(g.getNome());
        assertNull(g.getData());
    }

    @Test
    public void setNome_getNome_round_trip() {
        Grupo g = new Grupo();
        g.setNome("Familia");
        assertEquals("Familia", g.getNome());
    }

    @Test
    public void setId_getId_round_trip() {
        Grupo g = new Grupo();
        g.setId(42);
        assertEquals(42, g.getId());
    }

    @Test
    public void setData_getData_round_trip() {
        Grupo g = new Grupo();
        g.setData("25/12/2025");
        assertEquals("25/12/2025", g.getData());
    }

    @Test
    public void toString_retorna_nome() {
        Grupo g = new Grupo();
        g.setNome("Trabalho");
        assertEquals("Trabalho", g.toString());
    }

    @Test
    public void toString_com_nome_null_retorna_null() {
        Grupo g = new Grupo();
        assertNull(g.toString());
    }

    @Test
    public void serializable_round_trip_preserva_campos() throws Exception {
        Grupo original = new Grupo();
        original.setId(7);
        original.setNome("Amigos");
        original.setData("01/01/2025");

        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        ObjectOutputStream oos = new ObjectOutputStream(baos);
        oos.writeObject(original);
        oos.close();

        ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
        ObjectInputStream ois = new ObjectInputStream(bais);
        Grupo desserializado = (Grupo) ois.readObject();
        ois.close();

        assertEquals(original.getId(), desserializado.getId());
        assertEquals(original.getNome(), desserializado.getNome());
        assertEquals(original.getData(), desserializado.getData());
    }
}

File: app/src/test/java/activity/amigosecreto/db/GrupoModelTest.kt
package activity.amigosecreto.db

import org.junit.Assert.*
import org.junit.Test
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

class GrupoModelTest {

    @Test
    fun construtor_padrao_campos_sao_nulos_e_zero() {
        val g = Grupo()
        assertEquals(0, g.id)
        assertNull(g.nome)
        assertNull(g.data)
    }

    @Test
    fun setNome_getNome_round_trip() {
        val g = Grupo()
        g.nome = "Familia"
        assertEquals("Familia", g.nome)
    }

    @Test
    fun setId_getId_round_trip() {
        val g = Grupo()
        g.id = 42
        assertEquals(42, g.id)
    }

    @Test
    fun setData_getData_round_trip() {
        val g = Grupo()
        g.data = "25/12/2025"
        assertEquals("25/12/2025", g.data)
    }

    @Test
    fun toString_retorna_nome() {
        val g = Grupo()
        g.nome = "Trabalho"
        assertEquals("Trabalho", g.toString())
    }

    @Test
    fun toString_com_nome_null_retorna_string_vazia() {
        // Kotlin toString() retorna "" quando nome é null (String não pode ser null em Kotlin)
        val g = Grupo()
        assertEquals("", g.toString())
    }

    @Test
    fun serializable_round_trip_preserva_campos() {
        val original = Grupo()
        original.id = 7
        original.nome = "Amigos"
        original.data = "01/01/2025"

        val baos = ByteArrayOutputStream()
        ObjectOutputStream(baos).use { it.writeObject(original) }

        val desserializado = ObjectInputStream(ByteArrayInputStream(baos.toByteArray())).use {
            it.readObject() as Grupo
        }

        assertEquals(original.id, desserializado.id)
        assertEquals(original.nome, desserializado.nome)
        assertEquals(original.data, desserializado.data)
    }
}